std/algebra/native/sw_bls12377/pairing2.go

```diff
@@ -126,7 +126,12 @@ func (c *Curve) MultiScalarMul(P []*G1Affine, scalars []*Scalar) (*G1Affine, err
 	res := c.ScalarMul(P[0], scalars[0])
 	for i := 1; i < len(P); i++ {
 		q := c.ScalarMul(P[i], scalars[i])
-		c.Add(res, q)
+
```

Comment:
I think `ScalarMul` accounts for when `P[i]=(0,0)` no? It's `Add` that doesn't. Maybe use `AddUnified`?

Reply:
Yes, `Add` doesn't. Imo we do not have `AddUnified` for 2-chains, only for emulated?

Reply:
@yelhousni at some point you contemplated using a twisted Edwards model of the curve for this ? Is it still on the table ?